app/src/main/java/org/jellyfin/androidtv/details/EpisodeDetailsFragment.kt

```diff
@@ -33,23 +38,37 @@ class EpisodeDetailsFragment(private val data: Episode) : BaseDetailsFragment()
 			// Attach your media item details presenter to the row presenter:
 			val detailsDescriptionPresenter = DetailsDescriptionPresenter(title = data.name, subtitle = "TODO", body = data.description)
 
+			val overviewRowPresenter = FullWidthDetailsOverviewRowPresenter(
+				detailsDescriptionPresenter,
+				primaryImageBitmap?.let { DetailsOverviewLogoPresenter() })
+
+			overviewRowPresenter.onActionClickedListener = OnActionClickedListener {
+				val action = it as BaseAction
+				action.onClick()
+			}
 
-			addClassPresenter(DetailsOverviewRow::class.java,
-				FullWidthDetailsOverviewRowPresenter(
-					detailsDescriptionPresenter,
-					primaryImageBitmap?.let { DetailsOverviewLogoPresenter() }))
+			addClassPresenter(DetailsOverviewRow::class.java, overviewRowPresenter)
 
 
 			addClassPresenter(ListRow::class.java, ListRowPresenter())
 		}
 		rowsAdapter = ArrayObjectAdapter(selector)
 
-		val detailsOverview = DetailsOverviewRow("Media Item Details").apply {
-			imageDrawable = BitmapDrawable(resources, primaryImageBitmap)
-			// Add images and action buttons to the details view
-			addAction(Action(1, "Continue from 12:10"))
-			addAction(Action(2, "Play from Beginning"))
+		val actionsAdapter = SparseArrayObjectAdapter()
```

Comment:
I think an ArrayObjectAdapter works better so we don't need to add the integers for ordering (just make sure everything is added in the same place)

Reply:
Also, using the .apply or .also function is possible here too

Reply:
The documentation (recommended it)[https://developer.android.com/reference/android/support/v17/leanback/widget/DetailsOverviewRow], but for now it doesn't make much of a difference. Should I still change it to ArrayObjectAdapter?

Reply:
I think the docs recommend it because they are expecting the actions to be added/removed from various places. But we're doing it in a single place so a normal adapter should work fine.